src/scheduler/node_partition.c

```diff
@@ -579,7 +579,34 @@ update_buckets_for_node_array(node_bucket **bkts, node_info **ninfo_arr) {
 	for (i = 0; ninfo_arr[i] != NULL; i++) {
 		for (j = 0; bkts[j] != NULL; j++) {
 			int node_ind = ninfo_arr[i]->node_ind;
+
+			/* First turn off the current bit */
 			if (pbs_bitmap_get_bit(bkts[j]->bkt_nodes, node_ind)) {
+				if (pbs_bitmap_get_bit(bkts[j]->free_pool->truth, node_ind)) {
+					pbs_bitmap_bit_off(bkts[j]->free_pool->truth, node_ind);
+					bkts[j]->free_pool->truth_ct--;
+				} else if (pbs_bitmap_get_bit(bkts[j]->busy_later_pool->truth, node_ind)) {
+					pbs_bitmap_bit_off(bkts[j]->busy_later_pool->truth, node_ind);
+					bkts[j]->busy_later_pool->truth_ct--;
+				}  else if (pbs_bitmap_get_bit(bkts[j]->busy_pool->truth, node_ind)) {
+					pbs_bitmap_bit_off(bkts[j]->busy_pool->truth, node_ind);
+					bkts[j]->busy_pool->truth_ct--;
+				}
+
+				if (ninfo_arr[i]->num_jobs > 0 || ninfo_arr[i]->num_run_resv > 0) {
+					pbs_bitmap_bit_on(bkts[j]->busy_pool->truth, node_ind);
+					bkts[j]->busy_pool->truth_ct++;
+				} else {
+					if (ninfo_arr[i]->node_events != NULL) {
+						pbs_bitmap_bit_on(bkts[j]->busy_later_pool->truth, node_ind);
+						bkts[j]->busy_later_pool->truth_ct++;
+					} else {
+						pbs_bitmap_bit_on(bkts[j]->free_pool->truth, node_ind);
+						bkts[j]->free_pool->truth_ct++;
+					}
+
+				}
+				/* Next, turn on the correct bit */
 				if (ninfo_arr[i]->num_jobs > 0 || ninfo_arr[i]->num_run_resv > 0) {
```

Comment:
didn't you already turn on the correct bit in the code above?

Reply:
Good catch.  I had removed this code, but forgot to save the file.  It was sitting in my eclipse editor, but I submitted an older copy :(.  Silly me.  I've fixed it now.